components/clp-package-utils/clp_package_utils/scripts/archive_manager.py

```diff
@@ -0,0 +1,227 @@
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    find_parser.add_argument(
+        END_TS_ARG,
+        dest="end_ts",
+        type=int,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    # Options for delete subcommand
+    del_parser.add_argument(
+        DRY_RUN_ARG,
+        dest="dry_run",
+        action="store_true",
+        help="Only prints the archives to be deleted, without actually deleting them.",
+    )
+
+    # Subcommands for delete subcommand
+    del_subparsers = del_parser.add_subparsers(
+        dest="del_subcommand",
+        required=True,
+    )
+
+    # Delete by ID subcommand
+    del_id_parser = del_subparsers.add_parser(
+        BY_IDS_COMMAND,
+        help="Delete archives by ID.",
+    )
+
+    # Delete by ID arguments
+    del_id_parser.add_argument(
+        "ids",
+        nargs="+",
+        help="List of archive IDs to delete",
+    )
+
+    # Delete by filter subcommand
+    del_filter_parser = del_subparsers.add_parser(
+        BY_FILTER_COMMAND,
+        help="Delete archives that fall within the specified time range.",
+    )
+
+    # Delete by filter arguments
+    del_filter_parser.add_argument(
+        BEGIN_TS_ARG,
+        type=int,
+        default=0,
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    del_filter_parser.add_argument(
+        END_TS_ARG,
+        type=int,
+        required=True,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    parsed_args = args_parser.parse_args(argv[1:])
+
+    begin_timestamp = None
+    end_timestamp = None
+    subcommand = parsed_args.subcommand
+
+    # Validate and load config file
+    try:
+        config_file_path = Path(parsed_args.config)
+        clp_config = load_config_file(config_file_path, default_config_file_path, clp_home)
+        clp_config.validate_logs_dir()
+
+        # Validate and load necessary credentials
+        validate_and_load_db_credentials_file(clp_config, clp_home, False)
+    except:
+        logger.exception("Failed to load config.")
+        return -1
+
+    storage_type = clp_config.archive_output.storage.type
+    if StorageType.FS != storage_type:
+        logger.error(f"Archive deletion is not supported for storage type: {storage_type}.")
+        return -1
+
+    # Validate input depending on subcommands
+    if DEL_COMMAND == subcommand and BY_FILTER_COMMAND == parsed_args.del_subcommand:
+        begin_timestamp = parsed_args.begin_ts
+        end_timestamp = parsed_args.end_ts
+
+        # Validate the input timestamp
+        if not _validate_timestamps(begin_timestamp, end_timestamp):
+            return -1
+
+    elif FIND_COMMAND == subcommand:
+        begin_timestamp = parsed_args.begin_ts
+        end_timestamp = parsed_args.end_ts
+
+        if begin_timestamp is not None or end_timestamp is not None:
```

Comment:
Looking at the code again, `begin_timestamp` has a default value, so it should never gets a None value?

the `_validate_timestamps` is also written to handle None input, in this case, do we still need this `if begin_timestamp is not None or end_timestamp is not None:` statement?

If we can remove this if statement, maybe we can also combine the two if-clause for ts validation